### PR TITLE
Fix Npc trade

### PR DESCRIPTION
**Pull request by TheSumm**
Npc trade is not updating money / items if you move a container.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -3159,21 +3159,33 @@ void Player::postRemoveNotification(Thing* thing, const Cylinder* newParent, int
...
-		return;
+	uint32_t itemId = item->getID();
+	if (itemId != ITEM_GOLD_COIN && itemId != ITEM_PLATINUM_COIN && itemId != ITEM_CRYSTAL_COIN) {
+		auto it = std::find_if(shopItemList.begin(), shopItemList.end(), [itemId](const ShopInfo& shopInfo) { return shopInfo.itemId == itemId; });
+		if (it == shopItemList.end()) {
+			const Container* container = item->getContainer();
+			if (!container)
+				return false;
+
+			for (uint32_t index = 0; index < container->size(); ++index) {
```
Comment (djarek) [inline src/player.cpp]: if (std::any_of(container.begin(), container.end(), [] (Item*item) {return updateSaleShopList(item); })) {
return true;
}

File: src/player.cpp [hunk 2]
```diff
@@ -3159,21 +3159,33 @@ void Player::postRemoveNotification(Thing* thing, const Cylinder* newParent, int
...
 {
-	auto it = std::find_if(shopItemList.begin(), shopItemList.end(), [itemId] (const ShopInfo& shopInfo) { return shopInfo.itemId == itemId; });
-	if (it == shopItemList.end()) {
-		return;
+	uint32_t itemId = item->getID();
+	if (itemId != ITEM_GOLD_COIN && itemId != ITEM_PLATINUM_COIN && itemId != ITEM_CRYSTAL_COIN) {
+		auto it = std::find_if(shopItemList.begin(), shopItemList.end(), [itemId](const ShopInfo& shopInfo) { return shopInfo.itemId == itemId; });
+		if (it == shopItemList.end()) {
+			const Container* container = item->getContainer();
+			if (!container)
```
Comment (djarek) [inline src/player.cpp]: Missing curly bracket around return false;

**Comment by EPuncker** [comment]
great

**Comment by marksamman** [comment]
Thanks for the contribution!

**Pull request merged**